### ISSUE TITLE
Feature request: Provide multiple output formats to IJulia

### ISSUE THREAD

**Issue by benz0li** [JuliaPlots/Makie.jl#2195]
Currently, only one output format is provided: `png` by default; otherwise what is set by the `activate!` function (e.g. `CairoMakie.activate!(type = "svg")`, but only if there is a match with https://github.com/JuliaLang/IJulia.jl/blob/1cdaf7e5e65c1221b4a3fac86f63253bea65a0c5/src/display.jl#L19-L28.  

Please provide every MIME type specified in the first level of that vector.
👉 After executing `IJulia.register_mime(MIME("application/pdf"))` I would like `Makie.js` (`CairoMakie`) to return both `image/svg+xml` and `application/pdf`.

This allows `IJulia` to use
*  `image/svg+xml` for plot display in the browser
    \+ HTML conversion using SVG
*  `application/pdf` for direct PDF conversion

---

See https://github.com/JuliaPlots/Plots.jl/issues/4288 about the same feature request for `Plots.jl`.

**Comment by benz0li** [comment]
Cross references:

*  https://github.com/JuliaPlots/Plots.jl/pull/4188
    *  https://github.com/JuliaLang/IJulia.jl/issues/1035
    *  https://github.com/JuliaLang/IJulia.jl/pull/1038#issuecomment-1194523666
*  https://github.com/jupyter/docker-stacks/issues/1754  
    ℹ️ With `application/pdf` plot format in the notebook file, `inkscape` is not required for PDF conversion using `nbconvert`.

**Comment by jkrumbiegel** [comment]
One problem is that for some plots with lots of data points, vector graphics become very large and slow. In those cases bitmaps are desirable, but even though CairoMakie uses png by default, it seems that IJulia still saves an svg for every plot and displays that because it is ranked higher.

**Comment by benz0li** [comment]
> One problem is that for some plots with lots of data points, vector graphics become very large and slow. In those cases bitmaps are desirable,

Agreed. Execution time (Intel® Xeon® Silver 4210R CPU @ 2.40GHz) for https://lazarusa.github.io/BeautifulMakie/withTheme/theme_dark_astronauts/ [all packages installed and precompiled]:

*  png: 2m 32.1s (size: 1.1 MB)
*  svg: 14m 15s (size: 19.3 MB)
*  pdf: 2m 35s (size: 1.2 MB)

ℹ️ JupyterLab is able to display the svg while code-server fails miserably.

**Comment by benz0li** [comment]
> but even though CairoMakie uses png by default, it seems that IJulia still saves an svg for every plot and displays that because it is ranked higher.

I can't confirm that. You may login at https://demo.jupyter.b-data.ch with your GitHub account, Start image `Julia (pubtools:latest) + code-server` and execute the following as a Jupyter notebook (`.ipynb`)

```julia
Pkg.add("CairoMakie")
```

```julia
IJulia.ijulia_mime_types
```

```julia
import CairoMakie

x = range(0, 10, length = 100)
y = sin.(x)
CairoMakie.lines(x, y)
```

in JupyterLab or code-server; whatever you prefer.

If you then `cat` the `.ipynb` file in the Terminal, `cells.outputs.data` of the last cell contains `image/png` and `text/plain` only.  
ℹ️ IJulia does not do any image conversion on its own.

**Comment by SimonDanisch** [comment]
Feel free to open a PR with those changes! I don't think we currently have the resources to work on this!

**Comment by benz0li** [comment]
The issue is now described in detail and the solution a _nice to have_. I currently do not have the resources to fix this either.

I will see if I can contribute something at a later date.

P.S.: Makie is great!

**Comment by ffreyer** [comment]
Looks like you can set `type` in `CairoMakie.activate!()` now... Can this be closed?

**Comment by asinghvi17** [comment]
No, the question is asking for all possible formats to be marked showable/actually shown so Jupyter has access to png, svg and pdf output.
